src/Diagnostics.DataProviders/ChangeAnalysisClient.cs

```diff
@@ -0,0 +1,186 @@
+            return resourceChangesResponse;
+        }
+
+        /// <inheritdoc/>
+        public async Task<List<ChangeSetResponseModel>> GetChangeSetsAsync(ChangeSetsRequest changeSetsRequest)
+        {
+            string requestUri = changeAnalysisEndPoint + $"changesets?api-version={apiVersion}";
+            object postBody = new
+            {
+                changeSetsRequest.ResourceId,
+                StartTime = changeSetsRequest.StartTime.ToString(),
+                EndTime = changeSetsRequest.EndTime.ToString()
+            };
+            string jsonString = await PrepareAndSendRequest(requestUri, postBody, HttpMethod.Post);
+            List<ChangeSetResponseModel> changeSetsResponse = JsonConvert.DeserializeObject<List<ChangeSetResponseModel>>(jsonString);
+            return changeSetsResponse;
+        }
+
+        /// <inheritdoc/>
+        public async Task<ResourceIdResponseModel> GetResourceIdAsync(List<string> hostnames, string subscription)
+        {
+            string requestUri = changeAnalysisEndPoint + $"resourceId?api-version={apiVersion}";
+            object requestBody = new
+            {
+                hostNames = hostnames,
+                subscriptionId = subscription
+            };
+            string jsonString = await PrepareAndSendRequest(requestUri, requestBody, HttpMethod.Post);
+            if (!string.IsNullOrWhiteSpace(jsonString))
+            {
+                ResourceIdResponseModel resourceIdResponse = JsonConvert.DeserializeObject<ResourceIdResponseModel>(jsonString);
+                return resourceIdResponse;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Gets the last scan time stamp for a resource.
+        /// </summary>
+        /// <param name="armResourceUri">Azure Resource Uri.</param>
+        /// <returns>Last scan information.</returns>
+        public async Task<LastScanResponseModel> GetLastScanInformation(string armResourceUri)
+        {
+            string requestUri = changeAnalysisEndPoint + $"lastscan/{armResourceUri}?api-version={apiVersion}";
+            string jsonString = await PrepareAndSendRequest(requestUri, httpMethod: HttpMethod.Get);
+            return JsonConvert.DeserializeObject<LastScanResponseModel>(jsonString);
+        }
+
+        /// <summary>
+        /// Checks if a subscription has registered the ChangeAnalysis RP.
+        /// </summary>
+        /// <param name="subscriptionId">Subscription Id.</param>
+        public async Task<SubscriptionOnboardingStatus> CheckSubscriptionOnboardingStatus(string subscriptionId)
+        {
+            string requestUri = changeAnalysisEndPoint + $"Subscription/{subscriptionId}";
+            try
+            {
+               string jsonString = await PrepareAndSendRequest(requestUri, httpMethod: HttpMethod.Get);
+               return JsonConvert.DeserializeObject<SubscriptionOnboardingStatus>(jsonString);
+            }
+            catch (Exception)
```

Comment:
We should log the exception here.

Reply:
Use the data provider events here

Reply:
Correct me if i'm wrong. The Dataprovider calls this method and DataProvider method is already wrapped inside `DataProviderLogDecorator`. Should I add additional logging here?